instagram-app/src/components/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+const withAuthenticate = PostsPage => Login =>
+  class extends React.Component {
+    constructor() {
+      super();
+      this.state = {
+        loggedIn: false
+      };
+    }
+
+    componentDidMount() {
+      if (JSON.parse(localStorage.getItem("user"))) {
+        this.setState({
+          loggedIn: true
+        });
+      } else {
+        this.setState({
+          loggedIn: false
+        });
+      }
+    }
+    render() {
+      if (localStorage.getItem("User")) {
```

Comment:
```suggestion
      if (this.state.loggedIn) {
```
We should be checking the state here, not the local storage.